tests/clone_test.go

```diff
@@ -601,17 +610,21 @@ var _ = Describe("[Serial]VirtualMachineClone Tests", Serial, func() {
 					Expect(snapshotStorageClass).ToNot(BeEmpty(), "no storage class with snapshot support")
 				})
 
-				It("with a simple clone", func() {
+				It("with a simple clone, create clone before VM", func() {
 					running := false
 					if libstorage.IsStorageClassBindingModeWaitForFirstConsumer(snapshotStorageClass) {
 						// with wffc need to start the virtual machine
 						// in order for the pvc to be populated
 						running = true
 					}
-					sourceVM = createVMWithStorageClass(snapshotStorageClass, running)
+					sourceVM = generateVMWithStorageClass(snapshotStorageClass, running)
 					vmClone = generateCloneFromVM()
+					vmClone = createClone(vmClone)
 
-					createCloneAndWaitForFinish(vmClone)
+					events.ExpectEvent(vmClone, k8sv1.EventTypeNormal, "SourceDoesNotExist")
+					createVM(sourceVM, snapshotStorageClass, running)
+
+					waitCloneSucceeded(vmClone)
```

Comment:
WDYT about introducing another test for this specifically?

Reply:
We are discouraged from adding more tests if some scenarios can be embedded cleanly to an existing test. I think this small scenario of raciness between the creation of VM and Clone is pretty minor not sure its worth creating a test just for that and it only changes this simple happy path of creating a simple clone. WDYT?